### PR TITLE
[PHEE-641] Inconsistent behaviour Fix

### PR DESCRIPTION
**Pull request by abhinavm117**
## Description
Inconsistent behaviour Fix
Column Mismatch Fix
* PR title should have jira ticket enclosed in `[]`.<br>
Format: ``` [jira_ticket] description```<br>
ex: [phee-123] PR title.
* Add a link to the Jira ticket.
* Describe the changes made and why they were made.

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!
- [ ] Followed the PR title naming convention mentioned above.

- [ ] Design related bullet points or design document link related to this PR added in the description above.

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [ ] Created/updated unit or integration tests for verifying the changes made.

- [ ] Added required Swagger annotation and update API documentation with details of any API changes if applicable

- [ ] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

### CODE REVIEW

File: Dockerfile
Comment (fynmanoj) [inline Dockerfile:1]: remove unnecessary changes

File: src/main/java/org/mifos/processor/bulk/camel/routes/FileProcessingRoute.java
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/camel/routes/FileProcessingRoute.java:1]: remove unnecessary changes

File: src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java [hunk 1]
```diff
@@ -115,7 +115,6 @@ public void configure() throws Exception {
                     PaymentModeMapping mapping = paymentModeConfiguration.getByMode(paymentMode);
 
                     String tenantName = exchange.getProperty(TENANT_NAME, String.class);
-                    tenantName = mapping.getDebulkingDfspid() == null ? tenantName : mapping.getDebulkingDfspid();
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java:118]: why these lines are removed?

File: src/main/java/org/mifos/processor/bulk/camel/routes/SplittingRoute.java [hunk 2]
```diff
@@ -69,18 +72,39 @@ public void configure() throws Exception {
             List<String> subBatchFile = new ArrayList<>();
             Set<String> distinctPayeeIds = transactionList.stream().map(Transaction::getPayeeDfspId).collect(Collectors.toSet());
             logger.info("Payee id {}", distinctPayeeIds);
-            if (partyLookupEnabled && !distinctPayeeIds.isEmpty()) {
+            logger.info("Number of payeeId {}", distinctPayeeIds.size());
+            Boolean batchAccountLookup = (Boolean) exchange.getProperty("batchAccountLookup");
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/camel/routes/SplittingRoute.java:76]: use variable naming convention to differentiate a boolean variable

File: src/main/java/org/mifos/processor/bulk/camel/routes/SplittingRoute.java [hunk 3]
```diff
@@ -69,18 +72,39 @@ public void configure() throws Exception {
...
+                    subBatchIdList.add(subBatchId);
+                    subBatchIdPayeeMap.put(payeeId, subBatchId);
+                    subBatchIdMap.put(subBatchId, transactionsForPayee);
+                }
+                logger.info("Number of SubBatch based on payeeId {}", subBatchIdList.size());
+                // mapping subBatchId in transactionList
+                for (String subBatchId : subBatchIdList) {
+                    List<Transaction> transactions = subBatchIdMap.get(subBatchId);
+                    for (Transaction transaction : transactions) {
+                        for (Transaction originalTransaction : transactionList) {
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/camel/routes/SplittingRoute.java:100]: can we make this snippet more efficient?

File: src/main/java/org/mifos/processor/bulk/schema/Transaction.java [hunk 4]
```diff
@@ -32,6 +33,29 @@ public class Transaction implements CsvSchema {
...
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Transaction that = (Transaction) o;
+        return id == that.id && Objects.equals(requestId, that.requestId) && Objects.equals(paymentMode, that.paymentMode)
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/schema/Transaction.java:45]: this check looks very complicated and not readable, can you re-write

File: src/main/java/org/mifos/processor/bulk/utility/TransactionParser.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                transaction.setCycle(parts[11]);
+            }
+            if (parts.length > 13) {
+                transaction.setPayeeDfspId(parts[12]);
+            }
+
+            return transaction;
+        } catch (Exception e) {
+            logger.error("Error parsing line to Transaction object: {}", line, e);
+            return null;
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/utility/TransactionParser.java:63]: should we ignore the error here?

File: src/main/java/org/mifos/processor/bulk/zeebe/worker/InitSubBatchWorker.java [hunk 6]
```diff
@@ -72,15 +72,15 @@ public void setup() {
...
             SubBatchEntity subBatchEntity = null;
-
-            for (SubBatchEntity subBatch : subBatchEntityList) {
-                if (subBatch.getRequestFile().contains(fileName)) {
-                    subBatchEntity = subBatch;
-                    logger.info("SubBatchEntity found");
+            if (isSplittingEnabled) {
+                for (SubBatchEntity subBatch : subBatchEntityList) {
+                    if (subBatch.getRequestFile().contains(fileName)) {
+                        subBatchEntity = subBatch;
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/zeebe/worker/InitSubBatchWorker.java:78]: break this loop once the required element is found.
write a failure case if not found

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: Dockerfile
Comment (abhinavm117) [reply Dockerfile:1]: Its showing in git but nothing is changed, same command is written at line 5.

File: src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java (hunk 1 above)
Comment (abhinavm117) [reply src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java:118]: It was not used anywhere. 
Dfspid been used in line 120 directly                     
variables.put(DEBULKINGDFSPID, mapping.getDebulkingDfspid() == null ? tenantName : mapping.getDebulkingDfspid());

File: src/main/java/org/mifos/processor/bulk/camel/routes/FileProcessingRoute.java
Comment (abhinavm117) [reply src/main/java/org/mifos/processor/bulk/camel/routes/FileProcessingRoute.java:1]: Updated

File: src/main/java/org/mifos/processor/bulk/camel/routes/SplittingRoute.java (hunk 2 above)
Comment (abhinavm117) [reply src/main/java/org/mifos/processor/bulk/camel/routes/SplittingRoute.java:76]: Updated

File: src/main/java/org/mifos/processor/bulk/utility/TransactionParser.java (hunk 5 above)
Comment (abhinavm117) [reply src/main/java/org/mifos/processor/bulk/utility/TransactionParser.java:63]: Yes, if parsing fails we will log and continue to parse next transaction

File: src/main/java/org/mifos/processor/bulk/zeebe/worker/InitSubBatchWorker.java (hunk 6 above)
Comment (abhinavm117) [reply src/main/java/org/mifos/processor/bulk/zeebe/worker/InitSubBatchWorker.java:78]: updated

File: src/main/java/org/mifos/processor/bulk/camel/routes/SplittingRoute.java [hunk 7]
```diff
@@ -110,24 +126,40 @@ public void configure() throws Exception {
                 }
 
                 int subBatchCount = 1;
+                CsvSchema csvSchema = csvMapper.schemaFor(Transaction.class);
+                csvSchema = csvSchema.withHeader();
                 for (int i = 0; i < lines.size(); i += subBatchSize) {
+                    String subBatchId = UUID.randomUUID().toString();
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/camel/routes/SplittingRoute.java:132]: String formatters to be used for these usecases

**Review by fynmanoj — APPROVED** [review]
